File: 백준/Gold/1759. 암호 만들기/암호 만들기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.HashSet;
import java.util.List;
import java.util.Set;
import java.util.StringTokenizer;

public class Main {
    private static int L, C;
    private static char[] characters;
    private static final Set<Character> vowels = new HashSet<>(List.of('a', 'e', 'i', 'o', 'u'));
    private static StringBuilder print = new StringBuilder();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        L = Integer.parseInt(st.nextToken());
        C = Integer.parseInt(st.nextToken());

        characters = new char[C];
        characters = br.readLine().replaceAll(" ", "").toCharArray();

        Arrays.sort(characters);

        backtracking(0, "");
        System.out.println(print);
    }
    private static void backtracking(int index, String password) {
        if (password.length() == L
            && containsAtLeastOneVowel(password) && containsAtLeastTwoConsonant(password)) {
            print.append(password).append("\n");
        }
        if (index == C) return;
        if (password.length() < L) {
            backtracking(index + 1, password + characters[index]);
            backtracking(index + 1, password);
        }
    }
    private static boolean containsAtLeastOneVowel(String password) {
        for (char c: password.toCharArray()) {
            if (vowels.contains(c)) return true;
        }
        return false;
    }
    private static boolean containsAtLeastTwoConsonant(String password) {
        return password.chars()
                .filter(c -> !vowels.contains((char) c))
                .limit(2)
                .count() >= 2;
    }
}

File: 백준/Gold/1759. 암호 만들기/암호 만들기.cc
#include <iostream>
#include <vector>
#include <set>
#include <string>
#include <algorithm>

#define FAST_IO cin.tie(NULL)->ios::sync_with_stdio(false);

using namespace std;

#define REP(i, a, b) for (int i = a; i <= b; i++)

int l, c;
vector<char> characters;
set<char> vowels = { 'a', 'e', 'i', 'o', 'u' };
vector<string> output;

bool contains_at_least_one_vowel(const string &password) {
    for (char c : password) {
        if (vowels.count(c)) return true;
    }
    return false;
}

bool contains_at_least_two_consonant(const string &password) {
    int count = 0;
    for (char c : password) {
        if (!vowels.count(c)) {
            count++;
            if (count >= 2) return true;
        }
    }
    return false;
}

void backtracking(int index, const string &password) {
    if (password.size() == l) {
        if (contains_at_least_one_vowel(password) && contains_at_least_two_consonant(password)) {
            output.push_back(password);
        }
        return;
    }
    if (index == c) return;

    if (password.size() < l) backtracking(index + 1, password + characters[index]);

    backtracking(index + 1, password);
}

int main() {
    FAST_IO

    cin >> l >> c;
    characters.resize(c);

    REP(i, 0, c - 1) cin >> characters[i];
    sort(characters.begin(), characters.end());
    backtracking(0, "");

    for (string s : output) {
        cout << s << '\n';
    }

    return EXIT_SUCCESS;
}